src/adaXT/decision_tree/decision_tree.pyx

```diff
@@ -615,7 +666,8 @@ class DepthTreeBuilder:
                 root = new_node
             n_nodes += 1  # number of nodes increase by 1
 
-        tree.n_rows = X.shape[0]
+        tree.n_rows_fit = X.shape[0]
+        tree.n_rows_predict = X.shape[0]
```

Comment:
Shouldn't these be initialized at the beginning, so we can also document them as attributes?

Reply:
well they are only attributes once the tree has been fitted, no?

Reply:
Agreed but wouldnt it still be good to already initialize them at the top?